src/perf/lib/ThroughputClient.cpp

```diff
@@ -0,0 +1,409 @@
+/*++
+
+    Copyright (c) Microsoft Corporation.
+    Licensed under the MIT License.
+
+Abstract:
+
+    QUIC Perf Throughput Client Implementation.
+
+--*/
+
+#ifdef QUIC_CLOG
+#include "ThroughputClient.cpp.clog.h"
+#endif
+
+//
+// This needs to be included in kernel mode for QUIC_ADDR to work. Breaks user
+// mode because of QUIC_TEST_API, so only included in kernel mode
+//
+#ifdef _KERNEL_MODE
+#include <quic_platform.h>
+#endif
+#include "ThroughputClient.h"
+#include "ThroughputCommon.h"
+#include "quic_trace.h"
+
+ThroughputClient::ThroughputClient(
+    ) {
+    QuicZeroMemory(&LocalIpAddr, sizeof(LocalIpAddr));
+    if (Session.IsValid()) {
+        Session.SetAutoCleanup();
+    }
+}
+
+QUIC_STATUS
+ThroughputClient::Init(
+    _In_ int argc,
+    _In_reads_(argc) _Null_terminated_ char* argv[]
+    ) {
+    Port = THROUGHPUT_DEFAULT_PORT;
+    TryGetValue(argc, argv, "port", &Port);
+
+    const char* Target;
+    if (!TryGetValue(argc, argv, "target", &Target)) {
+        WriteOutput("Must specify '-target' argument!\n");
+        return QUIC_STATUS_INVALID_PARAMETER;
+    }
+
+    uint16_t Ip;
+    if (TryGetValue(argc, argv, "ip", &Ip)) {
+        switch (Ip) {
+        case 4: RemoteFamily = AF_INET; break;
+        case 6: RemoteFamily = AF_INET6; break;
+        }
+    }
+
+    TryGetValue(argc, argv, "length", &Length);
+    if (Length == 0) {
+        WriteOutput("Must specify a positive 'length'\n");
+        return QUIC_STATUS_INVALID_PARAMETER;
+    }
+
+    const char* LocalAddress;
+    if (TryGetValue(argc, argv, "bind", &LocalAddress)) {
+        if (!ConvertArgToAddress(LocalAddress, 0, &LocalIpAddr)) {
+            WriteOutput("Failed to decode IP address: '%s'!\nMust be *, a IPv4 or a IPv6 address.\n", LocalAddress);
+            return QUIC_STATUS_INVALID_PARAMETER;
+        }
+    }
+
+    // TODO: Core, since we need to support kernel mode
+    #ifdef QUIC_COMPARTMENT_ID
+    uint16_t CompartmentId;
+    if (TryGetValue(argc, argv, "comp",  &CompartmentId)) {
+        NETIO_STATUS status;
+        if (!NETIO_SUCCESS(status = QuicCompartmentIdSetCurrent(CompartmentId))) {
+            WriteOutput("Failed to set compartment ID = %d: 0x%x\n", CompartmentId, status);
+            return QUIC_STATUS_INVALID_PARAMETER;
+        } else {
+            WriteOutput("Running in Compartment %d\n", CompartmentId);
+        }
+    }
+
+    uint8_t CpuCore;
+    if (TryGetValue(argc, argv, "core",  &CpuCore)) {
+        QuicSetCurrentThreadAffinityMask((DWORD_PTR)(1ull << CpuCore));
+    }
```

Comment:
Don't include this in the comparment ID ifdef

Reply:
We don't have a linux equivalent yet, so this is a way to hide it.

Reply:
Just wrap it in an #ifdef QuicSetCurrentThreadAffinityMask